mito-ds/monorepo
# Import config improvements

# Description

- Adds the `decimal` configuration option to excel and csv imports
- Adds the `skiprows`configuration options to csv imports
- Refactors some of the simple import and excel import code to make adding params a bit easier in the future
- Logs user's pandas version as `version_pandas`
- Moves the package checking infrastructure to the frontend so we can just check it where we need it instead of having to continually add to the UserProfile.

Note: `decimal` was only added to read_excel in version 1.4 of pandas, so I only display that configuration option in the XLSX config taskpane when the user is post 1.4. 

# Testing

- See the tests
- Import files using `dev` and then switch to this branch and make sure imports still work
- Make sure that the `decimal` config is only available in the Excel import taskpane on or after pandas 1.4

Here is a csv and excel file with `,` decimals for testing:
[test_comma_decimal.xlsx](https://github.com/mito-ds/monorepo/files/9881195/test_comma_decimal.xlsx)
[test_comma_decimal.csv](https://github.com/mito-ds/monorepo/files/9881196/test_comma_decimal.csv)

- [ ] I have tested this on real data that is reasonable and large
- [ ] If I changed the interaction with JupyterLab, I tested that it does not break other programs (like VS Code), and tested that it works "multiple times" in the same notebook.

# Documentation

Note if any new documentation needs to addressed or reviewed.
